engine/core/src/main/java/org/datacleaner/job/tasks/CloseTaskListener.java

```diff
@@ -88,7 +93,10 @@ public void onComplete(Task task) {
 
     @Override
     public void onError(Task task, Throwable throwable) {
-        _success.set(false);
+        boolean alreadyRegisteredError = _errorsReported.getAndSet(true);
+        if (!alreadyRegisteredError) {
```

Comment:
Mostly because I find the meaning of the return of .getAndSet a little bit hard to infer. So naming it makes it clearer to read the if sentence: "if not already registered error".